subprojects/docs/src/docs/userguide/five_things.adoc

```diff
@@ -0,0 +1,125 @@
+
+[source,groovy]
+.Configuring a task within an action (execution phase) won't work
+----
+task copyFiles(type: Copy) {
+    doFirst {
+        into new Properties().load("my-props.properties").     // <1>
+                getProperty("notes.output.dir")                // <1>
+    }
+    from file("notes")                                         // <2>
+    include "**/*.txt"
+}
+----
+<1> Evaluated during _execution_ phase
+<2> Evaluated during _configuration_ phase
+
+This used to be more of a problem when the `<<` syntax for defining a task was more common, as it implicitly declared a task action — something that is evaluated/run during the execution phase. Users would often attempt to configure the task within that definition and wonder why it wasn't working.
+
+As a general rule of thumb, everything in a build script happens during the configuration phase unless it defines a task action. There are some other exceptions, such as link:{javadocPath}/org/gradle/api/execution/TaskExecutionGraph.html#whenReady-org.gradle.api.Action-[TaskExecutionGraph.whenReady(Action)], but the documentation is clear on their behavior.
+
+You can learn more in the <<build_lifecycle#build_lifecycle,Build Lifecycle>> chapter.
+
+== 4. Gradle is extensible in more ways than one
+
+It would be great if you could build your project using only the build logic bundled with Gradle, but that's rarely possible. Most builds have some special requirements that mean you need to add custom build logic.
+
+Gradle provides several mechanisms that allow you to extend it, from the very simple to the very powerful:
+
+ 1. Extra properties on <<writing_build_scripts#sec:extra_properties,projects>> and <<writing_build_scripts#sec:extra_task_properties,tasks>>.
++
+These allows you to add your own properties to a project or task that you can then use from your own custom actions or any other build logic. Extra properties can even be applied to tasks that aren't explicitly created by you, such as those created by Gradle's core plugins.
+
+ 2. Custom task actions.
++
+You can attach custom build logic that executes before or after a task via the link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doFirst(org.gradle.api.Action)[Task.doFirst()] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doLast(org.gradle.api.Action)[Task.doLast()] methods.
+
+ 3. <<custom_tasks,Custom task types>>.
++
+When you want the build to do some work that an existing task can't do, you can simply write your own task type. It's typically best to put the source file for a custom task type in the <<organizing_gradle_projects#sec:build_sources,_buildSrc_>> directory or in a packaged plugin. Then you can use the custom task type just like any of the Gradle-provided ones.
+
+ 4. Custom conventions.
```

Comment:
:x: We should not encourage the use of "conventions" to extend the DSL but of "extensions" instead.
See https://guides.gradle.org/implementing-gradle-plugins/#extensions_vs_conventions

Reply:
I'm talking about Gradle's internal conventions model, but generic conventions, such as directory structures and names. Since this chapter is primarily targeted at newcomers to Gradle, I'm pretty sure they will interpret "convention" in the traditional sense.

Reply:
Riiight, I mixed my mind :)